shared/src/validation-schema/user/user.ts

```diff
@@ -0,0 +1,23 @@
+import * as yup from 'yup';
+import 'yup-phone';
+import { RegisterPayloadKey, UserType, UserSex } from '../../common/enums'
+
+const validationUserSchema = yup.object().shape({
+  [RegisterPayloadKey.NAME]: yup.string().required(),
+  [RegisterPayloadKey.SURNAME]: yup.string().required(),
+  [RegisterPayloadKey.SEX]: yup.mixed<UserSex>().oneOf(Object.values(UserSex)),
+  [RegisterPayloadKey.BIRTH_DATE]: yup.date().required(),
+  [RegisterPayloadKey.EMAIL]: yup.string().required().email(),
+  [RegisterPayloadKey.PASSWORD]: yup.string().required().min(6),
+  [RegisterPayloadKey.RETYPE_PASSWORD]: yup.string().oneOf([yup.ref(RegisterPayloadKey.PASSWORD), null], 'Passwords must match'),
+  [RegisterPayloadKey.PHONE]: yup.string().required().phone(),
+  [RegisterPayloadKey.TYPE]: yup.mixed<UserType>().oneOf(Object.values(UserType)),
+  [RegisterPayloadKey.IMAGE_PATH]: yup.string().url(),
+  // [RegisterPayloadKey.AVATAR]: yup.mixed().defined().test(
+  //   "fileFormat",
+  //   "Unsupported Format",
+  //   value => (value.type === ContentType.PNG) || (value.type === ContentType.JPG)
+  // )
+});
```

Comment:
You need do add enums for messages and validation rule. See example: 
https://github.com/BinaryStudioAcademy/bsa-2021-healthcare/pull/4/files#diff-648aaf43dd4cd5c71efb4a95421b0c0c9987807674e9a9dac3840d96d12c3b9c
https://github.com/BinaryStudioAcademy/bsa-2021-healthcare/pull/4/files#diff-bccedc03a4e9853157271b61f94120856befb0d7197f628287222fd59ecc7b50